### DISCUSSION TITLE
Hello, is there Architecture design diagram of opengrok for studying?

### DISCUSSION THREAD

**Discussion by liaodalin19903** [oracle/opengrok#4175] [Q&A] [answered]
Hello, is there Architecture design diagram of opengrok for studying? 

I glance over the document of  opengrok, only find index data creation  diagram. is there any design diagram for opengrok?

https://i.stack.imgur.com/aMAyO.png  

---

By the way, the snapshot use opengrok.jar representative the processor, but in this article it should be indexer right? so whether the indexer is the most main component part of opengrok.jar?

**Comment by tulinkry** [comment, marked as answer]
It's pretty old image but it's still fairly accurate. Yes, the processor is called indexer as the main component and core of the opengrok and is included in the final opengrok.jar.

Additional diagram how the indexer works can be found here https://github.com/oracle/opengrok/wiki/Internals.

**Reply by vladak** [reply]
It depends on what level of detail you are interested in. There are surely some interesting design decisions not covered in the high level document.
